### PR TITLE
Feature commandforwarder

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: src/remixDClient.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  call (message, callback) {
+    const {action, id, key, name, payload} = message
+    try {
+      const svc = this.services[name]
+
+      if (name === 'fileManager') {
+        svc[key](payload, callback)
+      } else {
+        this.services["fileManager"]["createFolder"](this.sharedFolder) //Create folder if it doesn't exist
```
Comment (yann300) [inline src/remixDClient.js]: could we do that one in another PR if really needed?

File: src/services/sharedFolder.js [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
+const isbinaryfile = require('isbinaryfile')
+const fs = require('fs-extra')
+const chokidar = require('chokidar')
 
 module.exports = {
   trackDownStreamUpdate: {},
   websocket: null,
   alreadyNotified: {},
 
-  setWebSocket: function (websocket) {
```
Comment (yann300) [inline src/services/sharedFolder.js]: maybe this should be done separately

File: src/services/git.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  /**
+   * Create and check if command is valid
+   * @param action
+   * @param id
+   * @param key
+   * @param name
+   * @param payload
+   * @param callback
+   */
+  command (action, id, key, name, payload, callback) {
```
Comment (yann300) [inline src/services/git.js]: you don't need `action, id, key, name` in this function, at least you don't use it.
The signature can stay `command (payload, callback)`

File: src/remixDClient.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        svc.command(action, id, key, name, payload, callback)
```
Comment (yann300) [inline src/remixDClient.js]: Referring to a previous message, if you don't use `action, id, key, name`
This function could be simpler:

```
call (message, callback) {
    const {action, id, key, name, payload} = message
    try {
      const svc = this.services[name]
      svc[key](payload, callback)
    } catch (error) {
      callback({scope: name, key, error: error.message, result: null})
    }
  }
}
```

File: src/remixDClient.js (hunk 1 above)
Comment (edisinovcic) [reply src/remixDClient.js]: This has been removed and will be added to another PR.

File: src/services/sharedFolder.js (hunk 2 above)
Comment (edisinovcic) [reply src/services/sharedFolder.js]: Removed from this PR.

File: src/services/git.js (hunk 3 above)
Comment (edisinovcic) [reply src/services/git.js]: Currently sharedFolder service needs action, id, key, name and git and all others will use name, payload etc. Will have to check second part when remix-plugin is done. This way both services work.

File: src/remixDClient.js (hunk 4 above)
Comment (edisinovcic) [reply src/remixDClient.js]: We've updated callback to be in this format.
